src/components/RefinementList/RefinementList.tsx

```diff
@@ -1,37 +1,113 @@
 import Template from '../Template/Template';
 import RefinementListItem from './RefinementListItem';
 import SearchBox from '../SearchBox/SearchBox';
+import { RefinementListItem as TRefinementListItem } from '../../connectors/refinement-list/connectRefinementList';
+import { RefinementListOwnTemplates } from '../../widgets/refinement-list/refinement-list';
+import { RatingMenuTemplates } from '../../widgets/rating-menu/rating-menu';
+import { MenuTemplates } from '../../widgets/menu/menu';
+import { SearchBoxTemplates } from '../../widgets/search-box/types';
+import { CreateURL } from '../../types';
+
+export type RefinementListCSSClasses = {
+  depth?: string;
+  root?: string;
+  noRefinementRoot?: string;
+  list?: string;
+  item?: string;
+  selectedItem?: string;
+  parentItem?: string;
+  childList?: string;
+  searchBox?: string;
+  label?: string;
+  checkbox?: string;
+  labelText?: string;
+  count?: string;
+  noResults?: string;
+  showMore?: string;
+  disabledShowMore?: string;
+  disabledItem?: string;
+  searchable?: {
+    root: string;
+    form: string;
+    input: string;
+    submit: string;
+    submitIcon: string;
+    reset: string;
+    resetIcon: string;
+    loadingIndicator: string;
+    loadingIcon: string;
+  };
+};
+
+export type RefinementListProps = {
+  createURL: CreateURL<string>;
+  cssClasses?: RefinementListCSSClasses;
+  depth?: number;
+  facetValues?: TRefinementListItem[];
+  attribute?: string;
+  templateProps?:
+    | PreparedTemplateProps<RefinementListOwnTemplates>
+    | PreparedTemplateProps<RatingMenuTemplates>
+    | PreparedTemplateProps<MenuTemplates>;
+  searchBoxTemplateProps?: PreparedTemplateProps<SearchBoxTemplates>;
+  toggleRefinement: (value: string) => void;
+  searchFacetValues?: (query: string) => void;
+  searchPlaceholder?: string;
+  isFromSearch?: boolean;
+  showMore?: boolean;
+  toggleShowMore?: () => void;
+  isShowingMore?: boolean;
+  hasExhaustiveItems?: boolean;
+  canToggleShowMore?: boolean;
+  searchIsAlwaysActive?: boolean;
+  className?: string;
+  children?: h.JSX.Element;
+};
+
+const defaultProps = Object.freeze({
+  cssClasses: {},
+  depth: 0,
+});
+
+type RefinementListPropsWithDefaultProps = RefinementListProps &
+  typeof defaultProps;
+
+type RefinementListItemTemplateData = TRefinementListItem & {
+  url: string;
+} & Pick<RefinementListProps, 'attribute' | 'cssClasses' | 'isFromSearch'>;
 
-class RefinementList extends Component {
-  constructor(props) {
+class RefinementList extends Component<RefinementListPropsWithDefaultProps> {
+  public static defaultProps = defaultProps;
+
+  private searchBox = createRef<SearchBox>();
+
+  public constructor(props: RefinementListPropsWithDefaultProps) {
     super(props);
     this.handleItemClick = this.handleItemClick.bind(this);
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
-    const isStateDifferent = this.state !== nextState;
```

Comment:
why is this changed?

Reply:
This check was useless since the component has no internal state :wink: